packages/workspace/src/browser/workspace-delete-handler.ts

```diff
@@ -152,13 +152,53 @@ export class WorkspaceDeleteHandler implements UriCommandHandler<URI[]> {
         try {
             await Promise.all([
                 this.closeWithoutSaving(uri),
-                this.fileService.delete(uri, options)
+                this.deleteFile(uri, options)
             ]);
         } catch (e) {
             console.error(e);
         }
     }
 
+    /**
+     * Performs deletion of a given URI.
+     *
+     * @param uri URI of selected resource.
+     * @param options deletion options
+     */
+    protected async deleteFile(uri: URI, options: FileDeleteOptions): Promise<void> {
```

Comment:
The method has the same documentation as `delete`, perhaps elaborate on the differences between the two methods?

We can either:
- modify `delete` to perform the `catch` handling like you've done here
- rename the method to `deleteFilePermanently` and it will handle only the trash aspect.

Reply:
You are right that having the same documentation is a bit confusing. I think adapting the documentation of the `delete` method is more reasonable as it actually does two things: close without saving and deleting the file.

I think there is still value in having a separate `closeWithoutSaving` and `deleteFile` method whereas the new permanent deletion is only relevant for the `deleteFile`. For instance, if we move the error handling into the `delete` method the error could theoretically also come from the `closeWithoutSaving` (due to Promise.all) in which case the dialog would not make any sense. But maybe I misunderstood your suggestion.

I just wanted to raise my thoughts but I am not feeling too strongly about it, so if you want I can also change the PR into one of the other two options you provided.

Reply:
I see what you might mean, maybe we can document the method better and possibly rename it to make it clearer.